src/ui/win32/DialogBase.cpp

```diff
@@ -0,0 +1,128 @@
+#include "DialogBase.hh"
+
+#include "RA_Core.h" // g_RAMainWnd, g_hThisDLLInst
+
+namespace ra {
+namespace ui {
+namespace win32 {
+
+DialogBase::DialogBase(ra::ui::WindowViewModelBase& vmWindow) noexcept
+    : m_vmWindow(vmWindow), m_bindWindow(vmWindow)
+{
+}
+
+DialogBase::~DialogBase()
+{
+    if (m_hWnd)
+    {
+        DestroyWindow(m_hWnd);
+        m_hWnd = nullptr;
+    }
+}
+
+static INT_PTR CALLBACK StaticDialogProc(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    DialogBase* pDialog = reinterpret_cast<DialogBase*>(GetWindowLongPtr(hDlg, DWLP_USER));
+    if (pDialog == nullptr)
+        return DefWindowProc(hDlg, uMsg, wParam, lParam);
+
+    INT_PTR result = pDialog->DialogProc(hDlg, uMsg, wParam, lParam);
+
+    if (uMsg == WM_DESTROY)
+        SetWindowLongPtr(hDlg, DWLP_USER, 0L);
```

Comment:
In `WM_DESTROY `you're supposed to call `::PostQuitMessage(0)`.
What you have here is for `WM_NCDESTROY`.

Reply:
By the time `WM_NCDESTROY` is called, the `DWLP_USER` value is invalid which causes `pDialog->` to fail. This just says "after the WC_DESTROY message has been handled, stop listening to the remaining events." `DWLP_USER` will be nullptr, so line 26-27 will kick in for the remaining messages.

Reply:
I thought `PostQuitMessage` only applied to the main window. Doesn't it signal that the application is about to terminate?

Reply:
Thanks for that. I'm not certain but I think you're right. Dialogs have different rules than other window types.